colossalai/initialize.py

```diff
@@ -57,7 +58,7 @@ def launch(
     # set cuda device
     if torch.cuda.is_available() or IS_NPU_AVAILABLE:
         # if local rank is not given, calculate automatically
-        set_device(local_rank)
+        get_accelerator().set_device(local_rank)
```

Comment:
What about auto setting accelerator and detecting accelerator type here?

Reply:
ok